assembl/scripts/ini_files.py

```diff
@@ -144,72 +166,392 @@ def main():
-        assert exists(join(vroot, 'bin', 'virtuoso-t')),\
-            "virtuoso_root directory does not contain bin/virtuoso-t"
-        assert exists('var/db/virtuoso.ini.tmpl'),\
-            "Please run this script from the assembl root."
-        vroot_var = join(vroot, 'var')
-        if not exists(vroot_var):
-            vroot_var = '/var'
-        vroot_lib = join(vroot, 'lib')
-        assert exists(vroot_lib)
-        if not exists(join(vroot_lib, 'virtodbcu.so'))\
-                and exists(join(vroot_lib, 'odbc', 'virtodbcu.so')):
-            vroot_lib = join(vroot_lib, 'odbc')
-        vname = 'virtuoso'
-        if not exists(join(vroot, 'share', vname)):
-            names = listdir(join(vroot, 'share'))
-            names = [n for n in names
-                     if exists(join(vroot, 'share', n, 'vad'))]
-            assert len(names) == 1, "Cannot identify the vad directory"
-            vname = names[0]
-        assert exists(join(vroot_var, 'lib', vname, 'vsp')),\
-            "Cannot identify the VSP directory"
-        vars.update({
-            'VIRTUOSO_SERVER_PORT': config.getint('virtuoso', 'http_port'),
-            'VIRTUOSO_HOSTNAME': config.get(SECTION, 'public_hostname'),
-            'VIRTUOSO_PORT': config.getint('virtuoso', 'port'),
-            'VIRTUOSO_ROOT': vroot,
-            'VIRTUOSO_ROOT_VAR': vroot_var,
-            'VIRTUOSO_ROOT_LIB': vroot_lib,
-            'VIRTUOSO_SUBDIR_NAME': vname,
-        })
-    else:
-        # dummy values
-        vars.update({
-            'VIRTUOSO_SERVER_PORT': 8890,
-            'VIRTUOSO_HOSTNAME': 'localhost',
-            'VIRTUOSO_PORT': 1111,
-            'VIRTUOSO_ROOT': '/usr',
-            'VIRTUOSO_ROOT_VAR': '/usr/var',
-            'VIRTUOSO_ROOT_LIB': '/usr/var/lib',
-            'VIRTUOSO_SUBDIR_NAME': '/usr/share/virtuoso/vad',
-        })
-
-    print vars
-    for fname in ('var/db/virtuoso.ini', 'odbc.ini', 'supervisord.conf',):
+
+    for fname in ('supervisord.conf',):
         print fname
-        tmpl = open(fname+'.tmpl').read()
-        inifile = open(fname, 'w')
-        inifile.write(tmpl % vars)
-        inifile.close()
+        with open(fname + '.tmpl') as tmpl, open(fname, 'w') as inifile:
+            inifile.write(tmpl.read() % vars)
+
+
+def combine_ini(config, overlay, adding=True):
+    """Overlay values of the second ini file on the first.
+
+    Returns a ConfigParser object
+    If adding is false, only existing values will be replaced.
+    """
+    config = asParser(config)
+    overlay = asParser(overlay)
+    for section in overlay.sections():
+        # Avoid including DEFAULTSECT
+        for key in overlay._sections[section]:
+            value = overlay.get(section, key)
+            if adding or config.has_option(section, key):
+                ensureSection(config, section)
+                config.set(section, key, value)
+    if overlay._defaults:
+        config._defaults.update(overlay._defaults)
```

Comment:
Eh... the pythonic thing is to use private vars responsibly, being aware it is more prone to break. The responsible adult thing might be to add a test.... but since python2 is EOL, and python3 even seems to use the same var name, I'm not too worried.